src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/DotnetRestorePackageRestorer.cs

```diff
@@ -51,10 +62,13 @@ private Task<bool> RunRestoreAsync(IUpgradeContext context, string path, Cancell
             return _runner.RunProcessAsync(new ProcessInfo
             {
                 Command = "dotnet",
-                Arguments = Invariant($"restore \"{path}\""),
+                Arguments = _userInput.IsInteractive ? Invariant($"restore --interactive \"{path}\"") : Invariant($"restore \"{path}\""),
                 EnvironmentVariables = context.GlobalProperties,
                 Name = "dotnet-restore",
-                Quiet = true
+
+                // dotnet-restore does not need to receive user input to authenticate, so the only thing
+                // necessary to enable interactive auth is to make sure that necessary messages are displayed to users.
+                GetMessageLogLevel = message => MessagesToDisplay.Any(m => message.Contains(m, StringComparison.Ordinal)) ? LogLevel.Information : LogLevel.Debug,
```

Comment:
I understood why we would want to display 3 of the 4 messages. can we clarify why the message `*************************` was relevant?

Reply:
It's one of the lines that shows up in `dotnet restore`'s interactive auth prompt. It's just for formatting but leaving it out makes their prompt look less nice.